File: src/main/java/com/matamercer/microblog/models/entities/RefreshToken.java
package com.matamercer.microblog.models.entities;

import com.matamercer.microblog.models.entities.BaseModel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name= "refresh_tokens")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RefreshToken extends BaseModel {
    @ManyToOne
    @JoinColumn(name = "fk_user")
    private User user;

}

File: src/main/java/com/matamercer/microblog/models/entities/RefreshToken.kt
package com.matamercer.microblog.models.entities

import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "refresh_tokens")
class RefreshToken(
    @ManyToOne
    @JoinColumn(name = "fk_user")
    val user: User? = null
): BaseModel() {}